ThingIFSDK/ThingIFSDK/OnboardWithVendorThingIDOptions.swift

```diff
@@ -41,3 +41,32 @@ public struct OnboardWithVendorThingIDOptions {
         self.layoutPosition = position
     }
 }
+
+extension OnboardWithVendorThingIDOptions: JsonObjectCompatible {
+
+    internal init(_ jsonObject: [String : Any]) throws {
+        // This method may not use so this method is not tested.
```

Comment:
@yongpingchen This comes from negative reason. At first, `JsonObjectCompatible` was `JsonObjectSerializable` and `JsonObjectDeserialzable` in #364.
`JsonObjectSerializable` has `makeJsonObject`, and `JsonObjectDeserialzable` has `init`.

In the review, kumano-san pointed out that `JsonObjectSerialzable` and `JsonObjectDeserialzable` is bit strange. The comment is [this](https://github.com/KiiPlatform/thing-if-iOSSDK/pull/364#discussion_r104599030). I agreed with his opinion. We talked about the protocol names but we can not make proper names for each protocols.

Finally we unified the two protocols to `JsonObjectCompatible` protocol.

As the result of unifying the two protocol, we must implement the two methods in a class which adopts `JsonObjectCompatible`.

In some classes, one of the methods is meaningless but I and kumano san accepted the limitation.

This is the reason we provide that methods.

If you have any idea about the protocol names, Please tell me. I will and want to rename the protocol and divide it.